pkg/operation/care/seed_health.go

```diff
@@ -0,0 +1,86 @@
+package care
+
+import (
+	"context"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/dependencywatchdog"
+	seedpkg "github.com/gardener/gardener/pkg/operation/seed"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/go-logr/logr"
+)
+
+// SeedHealth contains information needed to execute health checks for seed.
+type SeedHealth struct {
+	seed       *gardencorev1beta1.Seed
+	seedClient kubernetes.Interface
+
+	logger logr.Logger
+}
+
+// NewHealthForSeed creates a new Health instance with the given parameters.
+func NewHealthForSeed(seed *gardencorev1beta1.Seed, seedClient kubernetes.Interface, logger logr.Logger) *SeedHealth {
+	return &SeedHealth{
+		seedClient: seedClient,
+		logger:     logger,
+		seed:       seed,
+	}
+}
+
+// CheckSeed conducts the health checks on all the given conditions.
+func (h *SeedHealth) CheckSeed(ctx context.Context,
+	seed *seedpkg.Seed,
+	conditions []gardencorev1beta1.Condition,
+	thresholdMappings map[gardencorev1beta1.ConditionType]time.Duration) []gardencorev1beta1.Condition {
+
+	var systemComponents gardencorev1beta1.Condition
+	for _, cond := range conditions {
+		switch cond.Type {
+		case gardencorev1beta1.SeedSystemComponentsHealthy:
+			systemComponents = cond
+		}
+	}
+
+	checker := NewHealthChecker(thresholdMappings, nil, nil, nil, nil)
+	newSystemComponents, err := h.checkSeedSystemComponents(ctx, checker, systemComponents)
+	return []gardencorev1beta1.Condition{NewConditionOrError(systemComponents, newSystemComponents, err)}
+}
+
+func (h *SeedHealth) checkSeedSystemComponents(
+	ctx context.Context,
+	checker *HealthChecker,
+	condition gardencorev1beta1.Condition,
+) (*gardencorev1beta1.Condition,
+	error) {
+
+	managedResources := managedResourcesSeed.List()
+
+	if h.seed.Spec.Settings.DependencyWatchdog != nil {
+		if h.seed.Spec.Settings.DependencyWatchdog.Endpoint != nil && h.seed.Spec.Settings.DependencyWatchdog.Endpoint.Enabled {
```

Comment:
I think there is already a helper func that does the same thing - `SeedSettingDependencyWatchdogEndpointEnabled`. You can check if you can use it.